core/src/net/network_channel.rs

```diff
@@ -104,13 +106,18 @@ impl TcpChannel {
         }
     }
 
-    pub fn get_id(&self) -> Option<Uuid> {
-        match self.state {
-            ChannelState::Connected(_, uuid) => Some(uuid),
-            ChannelState::Requested(_, uuid) => Some(uuid),
-            ChannelState::Initialised(_, uuid) => Some(uuid),
-            _ => None,
+    /// Checks the socket for errors and returns the current state
+    pub fn read_state(&mut self) -> &ChannelState {
+        match self.stream.take_error() {
```

Comment:
I love that all result of this function produce an error, whether they are `Ok` or `Err` :D

Reply:
It's like something I would write ^^